### ISSUE TITLE
.whereEqualTo("array", pointer) returns empty list below Lollipop (5.0)

### ISSUE THREAD

**Issue by ddlo86** [parse-community/Parse-SDK-Android#514]
With com.parse:parse-android:1.13.1, I'm unable to use the function whereEqualTo with pointers for devices less than API 20. Take my code below:

```
ParseQuery<JobDriver> driverQuery = ParseQuery.getQuery(JobDriver.class);
driverQuery.include("driver");
driverQuery.whereEqualTo("driver", ParseObject.createWithoutData(Driver.class, currentDriver.getObjectId()));
driverQuery.findInBackground(new FindCallback<JobDriver>() {
                        @Override
                        public void done(final List<JobDriver> drivers, ParseException e) {
                            if (e != null) {
                                Timber.e("Error finding driver: %s", e.getMessage());
                            } else {
                                Timber.d("Job size: %s", drivers.size());
```

With APIs 20 and above, the code works fine and produces a list of Drivers. But 4.4.4 and below returns an empty list (no error). 

I've found a work around for the moment by using whereContainedIn() and comparing the "driver" column to a List<Driver> with the single Driver object, where the "driver" column is an Array of Drivers.

JobDriver Table:
![image](https://cloud.githubusercontent.com/assets/471415/17794555/5c9b0118-6565-11e6-9bf9-4b7416ec2d92.png)

**Comment by patmelua** [comment]
Hi,

I've the same problem. Works great for API > 19 but not for API < 20. 
Any ideas ?

**Comment by felipesouto** [comment]
I had the same issue, but apparently the problem was solved by upgrading Parse Server version to 2.4.0 (see: https://github.com/parse-community/parse-server/issues/3169).
